core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -776,6 +776,11 @@ final object Decoder extends TupleDecoders with ProductDecoders with LowPriority
     new SemigroupK[Decoder] with MonadError[Decoder, DecodingFailure] {
       final def combineK[A](x: Decoder[A], y: Decoder[A]): Decoder[A] = x.or(y)
       final def pure[A](a: A): Decoder[A] = const(a)
+      final override def pureEval[A](a: Eval[A]): Decoder[A] = new Decoder[A] {
+        final def apply(c: HCursor): Result[A] = Xor.right(a.value)
```

Comment:
@johnynek I'm inclined to think that if there's a possibility that the expression you're pure-ing might not evaluate to anything, that's your responsibility, whether you're using `pure` or `pureEval` (and you should be using `instance` or whatever instead).